### DISCUSSION TITLE
Flow deck tof measurement equation

### DISCUSSION THREAD

**Discussion by siddarth-murugesan** [bitcraze/discussions#593] [Q&A]
Hi,
I am working on a project using crazyflie 2.1 and the Flow deck system. I came across the [mm_tof.c](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/kalman_core/mm_tof.c) function in the code, where the measurement equation was written as "h = z/((R*z_b)\dot z_b) = z/cos(alpha)". I couldn't understand it and couldn't find any documentation on it as well. Could you please explain the equation? What are the variables R and z_b in it?
Thank you

**Comment by knmcguire** [comment]
Yeah it is not super clear in the code... 

The initial implementation was done in the work of this master thesis: https://lup.lub.lu.se/student-papers/search/publication/8905295. Page 95, title 'Laser ranging measurement equations' should have your answer! But I think that using the z/cos(alpha) is already sufficient, so the extra calculation step was unnecessary.

I've made a ticket for your question so that we will be reminded to fix this: https://github.com/bitcraze/crazyflie-firmware/issues/1239. But... if you figure it all out anyway, I highly encourage you to do a pull request to the firmware 😄
